bats/auto_increment.bats

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+
+    dolt sql <<SQL
+CREATE TABLE test (
+  pk int NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  c0 int
+);
+SQL
+
+}
+
+teardown() {
+    teardown_common
+}
+
+@test "insert into auto_increment table" {
+    dolt sql -q "INSERT INTO test VALUES (1,11),(2,22),(3,33);"
+
+    run dolt sql -q "INSERT INTO test (c0) VALUES (44);"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "SELECT * FROM test WHERE c0 = 44;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "4,44" ]] || false
+
+    run dolt sql -q "INSERT INTO test (c0) VALUES (55),(66);"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "SELECT * FROM test WHERE c0 > 50;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "5,55" ]] || false
+    [[ "$output" =~ "6,66" ]] || false
+}
+
+@test "insert into empty auto_increment table" {
+    run dolt sql -q "INSERT INTO test (c0) VALUES (1);"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "SELECT * FROM test WHERE c0 = 1;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "1,1" ]] || false
+}
+
+@test "insert into auto_increment table with skipped keys" {
+    dolt sql -q "INSERT INTO test VALUES (1,1),(100,100);"
+
+    run dolt sql -q "INSERT INTO test (c0) VALUES (101);"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "SELECT * FROM test WHERE c0 > 100;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "101,101" ]] || false
+}
+
+@test "insert into auto_increment table with NULL" {
+    dolt sql -q "INSERT INTO test VALUES (1,1);"
+
+    run dolt sql -q "INSERT INTO test (pk,c0) VALUES (NULL,2);"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "SELECT * FROM test WHERE c0 > 1;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "2,2" ]] || false
+
+    run dolt sql -q "INSERT INTO test VALUES (NULL,3);"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "SELECT * FROM test WHERE c0 > 2;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "3,3" ]] || false
+}
+
+@test "insert into auto_increment table via batch mode" {
+    run dolt sql <<SQL
+INSERT INTO test (c0) VALUES (1);
+INSERT INTO test (c0) VALUES (2);
+INSERT INTO test (pk,c0) VALUES (19,19);
+INSERT INTO test (c0) VALUES (20);
+INSERT INTO test (c0) VALUES (21);
+SQL
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SELECT * FROM test;" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "1,1" ]] || false
+    [[ "$output" =~ "2,2" ]] || false
+    [[ "$output" =~ "19,19" ]] || false
+    [[ "$output" =~ "20,20" ]] || false
+    [[ "$output" =~ "21,21" ]] || false
+}
+
+@test "insert into table via batch mode" {
+    # asserts proper batch mode handling
+    run dolt sql <<SQL
+CREATE TABLE test2 (
+    pk int PRIMARY KEY,
+    c0 int
+);
+INSERT INTO test2 VALUES (1,1),(2,2),(3,3);
+INSERT INTO test2 SELECT (pk + 10), c0 FROM test2;
+INSERT INTO test2 SELECT (pk + 20), c0 FROM test2;
```

Comment:
This actually looks like a bug: the table should contain ((pk + 30), c0) rows, and it doesn't

Reply:
Or if this works, the test should assert it

Reply:
Looks like this should work based on a read of changes to the batching code